data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -82,15 +113,36 @@ public KeyValueProcessor(final PluginMetrics pluginMetrics, final KeyValueProces
                 throw new PatternSyntaxException("key_value_delimiter_regex is not a valid regex string", keyValueProcessorConfig.getKeyValueDelimiterRegex(), -1);
             }
 
+            if (keyValueProcessorConfig.getRecursive()) {
+                if (Pattern.compile(keyValueProcessorConfig.getKeyValueDelimiterRegex()).matcher(delimiterBracketCheck).matches()) {
```

Comment:
Similar as above, put L125 before this block and reuse `keyValueDelimiterPattern`